linkerd/stack/src/watch.rs

```diff
@@ -48,31 +47,50 @@ pub struct SpawnWatch<S> {
 // === impl NewSpawnWatch ===
 
 impl<P, N> NewSpawnWatch<P, N> {
-    pub fn new(new_update: N) -> Self {
+    pub fn new(inner: N) -> Self {
         Self {
-            new_update,
+            inner,
             _marker: std::marker::PhantomData,
         }
     }
+
+    pub fn layer() -> impl tower::layer::Layer<N, Service = Self> + Clone {
+        crate::layer::mk(Self::new)
+    }
+
+    // Creates a new `NewSpawnWatch` layer that transforms the target type to
+    // `T` for the inner stack.
```

Comment:
tiny nit: looks like these were supposed to be doc comments?
```suggestion
    /// Creates a new `NewSpawnWatch` layer that transforms the target type to
    /// `T` for the inner stack.
```